test/test.js

```diff
@@ -1,85 +1,48 @@
 'use strict';
-/* eslint-env mocha */
-var assert = require('assert');
-var execFile = require('child_process').execFile;
-var fs = require('fs');
-var path = require('path');
-var binCheck = require('bin-check');
-var BinBuild = require('bin-build');
-var compareSize = require('compare-size');
-var mkdirp = require('mkdirp');
-var rimraf = require('rimraf');
 
-var tmp = path.join(__dirname, 'tmp');
+const fs = require('fs');
+const path = require('path');
+const test = require('ava');
+const execa = require('execa');
+const tempy = require('tempy');
+const binCheck = require('bin-check');
+const BinBuild = require('bin-build');
+const compareSize = require('compare-size');
+const rimraf = require('rimraf');
+const pngquant = require('../');
 
-beforeEach(function (cb) {
-	mkdirp(tmp, cb);
-});
-
-afterEach(function (cb) {
-	rimraf(tmp, cb);
-});
-
-it('rebuild the pngquant binaries', function (cb) {
-	this.timeout(20000);
+test.cb('rebuild the pngquant binaries', t => {
+	const tmp = tempy.directory();
 
 	new BinBuild()
 		.src('https://github.com/pornel/pngquant/archive/2.7.1.tar.gz')
 		.cmd('rm ./INSTALL')
-		.cmd('./configure --prefix="' + tmp + '"')
-		.cmd('make install BINPREFIX="' + tmp + '"')
-		.run(function (err) {
-			if (err) {
-				cb(err);
-				return;
-			}
-
-			fs.stat(path.join(tmp, 'pngquant'), function (err, stats) {
-				if (err) {
-					cb(err);
-					return;
-				}
-
-				assert(stats.isFile());
-				cb();
-			});
+		.cmd(`./configure --prefix="${tmp}"`)
+		.cmd(`make install BINPREFIX="${tmp}"`)
+		.run(err => {
+			t.ifError(err);
+			t.true(fs.existsSync(path.join(tmp, 'pngquant')));
+			rimraf.sync(tmp);
```

Comment:
No need to do this. The system will handle cleaning up stuff in the temp directory. That's why it's called a temp directory.